### ISSUE TITLE
[Enhancement] Support external TURN with static auth secret

### ISSUE THREAD

**Issue by Sommerwiesel** [schlagmichdoch/PairDrop#348]
**What problem is solved by the new feature**
Increased security
Support modern turn servers

**Describe the feature**
The files and doc on this github only tell about how to use an external TURN server that uses the ancient, by coturn unrecommended lt-cred mechanism (username + secret).

My coturn runs in (according to coturn docs officially recommended) static-auth-secret mode.

Sadly, I cannot figure out how to make pairdrop use my coturn as TURNS Server with static auth secret.
If I omit the username in the rtc json pairdrop throws an error that it needs an username.
This makes me think pairdrop doesn't even support static-auth secret.

**Please support static-auth secret for external turn.**
You should possible also deprecate the old lt-cred authentification.
It's neither safe nor recommended.

In case it's already supported, please document on how to set it up using the rtc_config.json, because I can't seem to figure it out on my own.

**Comment by schlagmichdoch** [comment]
You are correct that currently only static credentials are supported via the lt-cred mechanism as this was the easiest to implement to begin with. Beware, that any information you put into `rtc-config.json` will be public as its content is sent to clients as is to be used as the configuration of all RTCPeerConnection.

It's a good idea to implement expiring credentials via the static-auth-secret mode in the future but this requires a new API to derive credentials from the static-auth-secret and some logic to restart the signaling process once the credentials are expired (e.g. after 24h?). In Firefox the [setting of configurations of existing RTCPeerConnections](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/setConfiguration) is only supported since 2022 though so for older versions we might have to come up with a polyfill which completely reinstantiates existing RTCPeerConnections or tells the user to reload the page.

I guess the creation of credentials would then look similar to this: https://stackoverflow.com/a/35767224/14678591
The username will then include the timestamp when the credentials expire. As an admin you will then only have to set the static-auth-secret via a new env var like `RTC_STATIC_AUTH_SECRET` or similar.

Beware though that there are lots of issues with higher priority so this will probably not be deployed until `v2.2`.

**Comment by Espo02** [comment]
Hi
Thank you for all the work which is done. 
I found this post because of I'm hosting a turn server for nextcloud and synapse which both of them are using "RTC_STATIC_AUTH_SECRET". So I can't configure my server to username and password for Pairdrop.
It would be very helpful to me if this option would be added.

Best regards,
Michael

**Comment by brknkfr** [comment]
I just stumbled upon the same problem. You can circumvent the problem by using this small shell script and create a `username` and a `password` by using the `shared_secret` of your existing coturn.

```
ttl=31557600 # A year
secret=<YourCoturnSharedSecret>
timestamp=$(( $(date +%s) + ttl ))
username=$timestamp:$user
digest=$(printf '%b' "$username" | /usr/bin/openssl dgst -sha1 -hmac "$secret" -binary)
password=$(printf '%b' "$digest" | base64)
echo "username: $username"
echo "password: $password"

```
(This was taken and adapted from https://wiki.lenuagemagique.com/doku.php?id=unable_to_use_lt-cred-mech_webrtc_and_static-auth-secret_restapi_at_the_same_time.)

Then just use a long `ttl`, so that it doesn't expire immediately and don't forget to test the credentials with `turnutils_uclient -y -u "$username" -w "$password" turnserver.domain.tld`.

**Comment by EnigmaCurry** [comment]
As @brknkfr mentioned, it works just fine with short term credentials, but you will have to figure out your own way of rotating them. From my reading of the pairdrop source code, it will only read the rtc_config.json one time on startup, and it doesn't watch that file for changes. So, my solution was to make the container mint new credentials on container startup and on a schedule, and then to automatically restart pairdrop when that happens.

In my docker container's config, I set PAIRDROP_RESTART_MINUTES=240 so the service will restart every 4 hours. The credentials TTL I have set to PAIRDROP_RESTART_MINUTES * 2 (8 hours), so even clients that get tokens right before the server restarts will still work for yet another 4 hours. The pairdrop client itself already seems to handle the reconnection to the restarted server fine.

Here's my config: https://github.com/EnigmaCurry/d.rymcg.tech/tree/master/pairdrop
